### DISCUSSION TITLE
FAVORITEの挙動の改善について

### DISCUSSION THREAD

**Discussion by Shimi9999** [exch-bms2/beatoraja#676] [General]
現在のFAVORITEキー(F8/F9)はキーを押す度に `FAVORITE→INVISIBLE→なし` という挙動をしますが、INVISIBLEを経由するのは不便さや分かりにくさを生んでいるように感じます。

問題
- INVISIBLEの存在を知らないと、FAVORITEを解除しようとしてフォルダから曲が消えて混乱する。
- FAVORITE解除時にINVISIBLEを経由するせいで、毎回INVISIBLEフォルダまで探しに行かないといけない。

これを解決するためには、FAVORITEとINVISIBLEのオンオフは別々にすべきではないでしょうか。
- キー単押しならFAVORITE、キー長押しならINVISIBLE
- 通常はFAVORITE、DELETEキー押しながらでINVISIBLE

など。

**Comment by wcko87** [comment]
今Ctrl,Shiftとかできるので(#702)それを利用するのはどうでしょうか？
F8 → FAVORITE
Ctrl+F8 → INVISIBLE

**Reply by Shimi9999** [reply]
CtrlとShiftは初期状態だとスクラッチ操作が割り当てられているので、操作が競合する懸念があり、個人的には不適かなと思っています。
